packages/dev-utils/src/devApp/render.tsx

```diff
@@ -240,7 +235,9 @@ export class DevAppBuilder {
       window.location.pathname = this.defaultPage;
     }
 
-    ReactDOM.render(<DevApp />, document.getElementById('root'));
+    const container = document.getElementById('root');
+    const root = createRoot(container!);
+    root.render(<DevApp />);
```

Comment:
Since we've got a peer dep with a range that can resolve to either 17 or 18 we'll need to do feature detection here. Basically `ReactDom.createRoot || ReactDom.render`. While `render` is still available in 18 you'll get a warning if you use it, so I think we should properly fall back instead. That goes for the changes later on in this PR too.

Reply:
That makes sense